pkg/minikube/cluster/cluster.go

```diff
@@ -88,9 +88,9 @@ func CacheISO(config cfg.MachineConfig) error {
 
 // StartHost starts a host VM.
 func StartHost(api libmachine.API, config cfg.MachineConfig) (*host.Host, error) {
```

Comment:
I find our inconsistent naming of `cfg` vs `config` absolutely confusing, mostly because it violates the following Go tenet: "Avoid renaming imports except to avoid a name collision; good package names should not require renaming. In the event of collision, prefer to rename the most local or project-specific import.": https://github.com/golang/go/wiki/CodeReviewComments#imports

For a follow-up PR, we should more consistently name our config references. I suggest:

-  `k8s.io/minikube/pkg/minikube/config` should be referred to as `config`
- Types defined in `pkg/minikube/config` should not stutter: perhaps `config.Profile`, `config.Machine`, `config.Kubernetes`?
- `cfg` currently can refer to `config.Config`, `config.MachineConfig`, or `config.KubernetesConfig`. We should find more consistent identifiers based on whatever the non-stuttering names are.

I won't make you do this in this PR however. Perhaps before you get to the point that a profile (or `config.Config`) object needs to refer to multiple machines though :)

Reply:
I followed up with @sharifelgamal and there will be a follow-up PR that addresses this and some concerns that @medyagh had:

- An object (named something like config.Cluster or config.Profile) will contain multiple config.Machine objects.
- config.Machine will contain a single config.Kubernetes object associated to that machine

This will allow us to handle multi-node and drivers that need additional Kubernetes hints like the version, as well as give us room to configure kubelet differently on the master versus worker nodes.